include/mfmg/adapters_dealii.hpp

```diff
@@ -0,0 +1,199 @@
+/*************************************************************************
+ * Copyright (c) 2017-2018 by the mfmg authors                           *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * This file is part of the mfmg libary. mfmg is distributed under a BSD *
+ * 3-clause license. For the licensing terms see the LICENSE file in the *
+ * top-level directory                                                   *
+ *                                                                       *
+ * SPDX-License-Identifier: BSD-3-Clause                                 *
+ *************************************************************************/
+
+#ifndef MFMG_ADAPTERS_DEALII_HPP
+#define MFMG_ADAPTERS_DEALII_HPP
+
+#include <boost/property_tree/ptree.hpp>
+
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/lac/trilinos_solver.h>
+
+#include "amge_device.cuh"
```

Comment:
This is the reason why the tester fails. `amge_host.cc` include `adapters_dealii.hpp` which include `amge_device.cuh` which contains cuda code. Since `amge_host.cc` is sent to `gcc` it cannot find `cusparse.h`. I don't know why it only fails in docker but it works on native install.

Reply:
Oh yeah, this is historical artifact. Originally, I was trying to put both device and host things in Adapter, but since then I removed the `Device`. Thanks for catching that!